src/main/scala/passes/ReplaceRsvdKeywords.scala

```diff
@@ -0,0 +1,103 @@
+package essent.passes
+
+import firrtl.{Transform, WRef}
+import firrtl.constraint.IsVar
+import firrtl.ir._
+import firrtl.passes.Pass
+
+object ReplaceRsvdKeywords extends Pass {
+  val reservedWords = Set("asm" ,	"else",	"new",	"this",
+    "auto",	"enum",	"operator",	"throw",
+    "bool",	"explicit",	"private",	"true",
+    "break",	"export",	"protected",	"try",
+    "case",	"extern",	"public",	"typedef",
+    "catch",	"false",	"register",	"typeid",
+    "char",	"float",	"reinterpret_cast",	"typename",
+    "class",	"for",	"return",	"union",
+    "const",	"friend",	"short",	"unsigned",
+    "const_cast",	"goto",	"signed",	"using",
+    "continue",	"if",	"sizeof",	"virtual",
+    "default",	"inline",	"static",	"void",
+    "delete",	"int",	"static_cast",	"volatile",
+    "do",	"long",	"struct",	"wchar_t",
+    "double",	"mutable",	"switch",	"while",
+    "dynamic_cast",	"namespace",	"template",
+    "And",	"bitor",	"not_eq",	"xor",
+    "and_eq",	"compl",	"or",	"xor_eq",
+    "bitand",	"not",	"or_eq", "UNLIKELY",
+    "PARTflags", "update_registers","verbose",
+    "done_reset" , "sim_cached","regs_set",
+    "assert_triggered","assert_exit_code")
+
+  override def prerequisites = Seq.empty
+  override def optionalPrerequisites = firrtl.stage.Forms.LowFormOptimized
+  override def invalidates(a: Transform) = false
+  private def renameWord(oldName: String): String = oldName + "$"
+  private def maybeRename(oldName: String): String =
+    if (reservedWords.contains(oldName)) renameWord(oldName)
+    else oldName
+
+  private def analyzeExpression(e: Expression): Expression = e match {
+    case r: Reference => r.copy(name = maybeRename(r.name))
+    case r: WRef => r.copy(name = maybeRename(r.name))
+    case m: Mux => m.mapExpr(analyzeExpression)
+    case s: SubField => s.copy(name = maybeRename(s.name)).mapExpr(analyzeExpression)
+    case d: DoPrim => d.mapExpr(analyzeExpression)
+    case si: SubIndex => si.mapExpr(analyzeExpression)
+    case sa: SubAccess => sa.mapExpr(analyzeExpression)
+    case vi: ValidIf => vi.mapExpr(analyzeExpression)
+    case _ => e
+  }
+
+  private def analyzeIsVar(iv: IsVar):IsVar = iv match {
```

Comment:
Is support for IsVar necessary? Is this called from anywhere?

Reply:
IsVar is related to the constraint solver (determining width) and isn't necessary here since it's used internally in a FIRRTL pass , so I have removed and run the test